lib/screens/scan_screen.dart

```diff
@@ -0,0 +1,89 @@
+import 'dart:io';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+
+class ScanScreen extends StatefulWidget {
+  const ScanScreen({Key? key}) : super(key: key);
+
+  @override
+  State<ScanScreen> createState() => _ScanScreenState();
+}
+
+class _ScanScreenState extends State<ScanScreen> {
+  final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+  Barcode? result;
+  QRViewController? controller;
```

Comment:
**suggestion**: to avoid working with nullable types, you could have either:

  - initialized this as `late QRViewController controller`
  - give it a default value (if possible) like `QRViewController controller = QRViewController()` or similar, if any